common/src/lib.rs

```diff
@@ -400,29 +402,41 @@ pub fn save_page_lazy_info(id: H256, page_num: PageNumber) {
     gear_ri::gear_ri::save_page_lazy_info(page_num.0, &key);
 }
 
-pub fn get_program_pages(
+pub fn get_program_pages_data(id: H256, program: &ActiveProgram) -> BTreeMap<PageNumber, Vec<u8>> {
+    get_program_data_for_pages(id, program.pages_with_data.iter())
+}
+
+/// Returns data for all pages from `pages` arg, which has data in storage.
+pub fn get_program_data_for_pages<'a>(
     id: H256,
-    pages: BTreeSet<PageNumber>,
-) -> Option<BTreeMap<PageNumber, Vec<u8>>> {
-    let mut persistent_pages = BTreeMap::new();
-    for page_num in pages {
-        let key = page_key(id, page_num);
+    pages: impl Iterator<Item = &'a PageNumber>,
+) -> BTreeMap<PageNumber, Vec<u8>> {
+    pages
+        .map(|p| {
+            let key = page_key(id, *p);
+            (*p, sp_io::storage::get(&key))
+        })
+        .filter_map(|(page, data)| data.map(|data| (page, data)))
+        .collect()
+}
 
-        persistent_pages.insert(page_num, sp_io::storage::get(&key)?);
-    }
-    Some(persistent_pages)
+pub fn set_program(id: H256, program: ActiveProgram) {
```

Comment:
I would mark it as non public since it looks like it doesn't get used anywhere except this crate.

Reply:
There are uses